### ISSUE TITLE
Introduce unit tests for branch protection

### ISSUE THREAD

**Issue by Necr0x0Der** [trueagi-io/metta-morph#16]
Unit tests should systematically cover various aspects of MeTTa-Scheme interactions and run automatically for branch protection on github. Not working cases could possibly exist as skipped unit tests. More complex examples can remain as integration tests.

**Comment by patham9** [comment]
I agree. It can be initially based on example1...6.metta in ./extend: Running automatically same file, once compiling the "to-compile-code" with compiler, one time also running the "to-compile-code" with the MeTTa interpreter, then checking whether the output matches. (same principle as in ./tests but this time MeTTa interpreter is involved in both cases). I can add a script which does this soon.

**Comment by patham9** [comment]
https://github.com/trueagi-io/metta-morph/pull/21
